osu.Game/Overlays/Music/PlaylistItem.cs

```diff
@@ -20,118 +18,96 @@
 
 namespace osu.Game.Overlays.Music
 {
-    public partial class PlaylistItem : OsuRearrangeableListItem<Live<BeatmapSetInfo>>, IFilterable
+    public partial class PlaylistItem : PoolableDrawable, IHasCurrentValue<Live<BeatmapSetInfo>>
     {
-        public readonly Bindable<Live<BeatmapSetInfo>> SelectedSet = new Bindable<Live<BeatmapSetInfo>>();
+        public Bindable<Live<BeatmapSetInfo>> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
 
-        public Action<Live<BeatmapSetInfo>> RequestSelection;
+        private readonly BindableWithCurrent<Live<BeatmapSetInfo>> current = new BindableWithCurrent<Live<BeatmapSetInfo>>();
 
-        private TextFlowContainer text;
-        private ITextPart titlePart;
+        private readonly Bindable<Live<BeatmapSetInfo>?> selectedSet = new Bindable<Live<BeatmapSetInfo>?>();
+        private Action<Live<BeatmapSetInfo>>? requestSelection;
 
-        [Resolved]
-        private OsuColour colours { get; set; }
+        private OverflowScrollingContainer text = null!;
 
-        public PlaylistItem(Live<BeatmapSetInfo> item)
-            : base(item)
-        {
-            Padding = new MarginPadding { Left = 5 };
-        }
+        [Resolved]
+        private OsuColour colours { get; set; } = null!;
 
         [BackgroundDependencyLoader]
-        private void load()
+        private void load(PlaylistOverlay playlistOverlay)
         {
-            HandleColour = colours.Gray5;
+            RelativeSizeAxes = Axes.X;
+            Height = 20;
+
+            InternalChild = text = new OverflowScrollingContainer
```

Comment:
I'm not sold on every long item always scrolling. Probably better only when hovered?

Reply:
I tried, please see how you find https://github.com/ppy/osu/pull/32679/commits/b8360a19dd96a18075821c056f4c6723b4d3379e (especially the behaviour when un-hovering mid-scroll).

Reply:
I'd expect the scroll to reset to original position on unhovering.

Reply:
That's how I had it initially and it was a much simpler implementation. Guess I'll roll back...